llvm/lib/Transforms/InstCombine/InstCombineCasts.cpp

```diff
@@ -469,10 +473,23 @@ static Instruction *foldVecExtTruncToExtElt(TruncInst &Trunc,
   uint64_t BitCastNumElts = VecElts.getKnownMinValue() * TruncRatio;
   uint64_t VecOpIdx = Cst->getZExtValue();
   uint64_t NewIdx = IC.getDataLayout().isBigEndian()
-                        ? (VecOpIdx + 1) * TruncRatio - 1
+                        ? (VecOpIdx * TruncRatio) + (TruncRatio - 1)
```

Comment:
Isn't it equivalent to the original code?

Reply:
Yes, in an earlier draft I was trying to get the DAG/InstCombine implementations to match - I've undone the change